### PR TITLE
Add network settings for connect and read timeouts when fetching Build Scan data

### PR DESCRIPTION
**Pull request by jthurne**
- Add an ability to configure build scan fetch timeout
- Represent connect and read timeouts as Durations

### CODE REVIEW

File: components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/api/client/GradleEnterpriseApiClient.java [hunk 1]
```diff
@@ -99,6 +100,27 @@ private boolean allowUntrustedServer() {
...
+    }
+
+    private Duration parseTimeout(String key) {
+        String value = System.getProperty(key);
+        if (value == null) {
+            // value is not set
+            return null;
+        } else {
+            return Duration.parse(key);
+        }
```
Comment (etiennestuder) [inline components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/api/client/GradleEnterpriseApiClient.java:121]: I would use the ternary operator here to reduce this if/else to a single line.

File: components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/api/client/GradleEnterpriseApiClient.java [hunk 2]
```diff
@@ -99,6 +100,27 @@ private boolean allowUntrustedServer() {
...
+    }
+
+    private Duration parseTimeout(String key) {
+        String value = System.getProperty(key);
+        if (value == null) {
+            // value is not set
+            return null;
+        } else {
+            return Duration.parse(key);
```
Comment (etiennestuder) [inline components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/api/client/GradleEnterpriseApiClient.java:120]: You are parsing the key not the value.

**Review by etiennestuder — CHANGES_REQUESTED** [review]

File: components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/network/NetworkSettingsConfigurator.java [hunk 3]
```diff
@@ -43,6 +43,10 @@ private static boolean isProxyProperty(String key) {
             || key.endsWith(".nonProxyHosts");
     }
 
+    private static boolean isTimeoutProperty(String key) {
```
Comment (etiennestuder) [inline components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/network/NetworkSettingsConfigurator.java:46]: Maybe rename the method to `isConnectionProperty`?

File: components/scripts/network.settings [hunk 4]
```diff
@@ -17,3 +17,7 @@
 
 # Uncomment to disable SSL certificate validation when connecting to the Gradle Enterprise server
 #ssl.allowUntrustedServer=true
+
+# Uncomment to use a custom connect or read timeout when fetching Build Scan data.
+#connect.timeout=PT30S
```
Comment (etiennestuder) [inline components/scripts/network.settings:22]: I would reduce this example to 5s.

**Review by etiennestuder — CHANGES_REQUESTED** [review]

File: components/scripts/network.settings [hunk 5]
```diff
@@ -17,3 +17,7 @@
 
 # Uncomment to disable SSL certificate validation when connecting to the Gradle Enterprise server
 #ssl.allowUntrustedServer=true
+
+# Uncomment to use a custom connect or read timeout when fetching Build Scan data.
```
Comment (etiennestuder) [inline components/scripts/network.settings:21]: Remove the dot at the end of the sentence

**Review by etiennestuder — CHANGES_REQUESTED** [review]

File: components/scripts/network.settings (hunk 5 above)
Comment (jthurne) [reply components/scripts/network.settings]: ✅ Done

File: components/scripts/network.settings (hunk 4 above)
Comment (jthurne) [reply components/scripts/network.settings]: ✅ Done

File: components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/network/NetworkSettingsConfigurator.java (hunk 3 above)
Comment (jthurne) [reply components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/network/NetworkSettingsConfigurator.java]: ✅ Done

File: components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/api/client/GradleEnterpriseApiClient.java (hunk 1 above)
Comment (jthurne) [reply components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/api/client/GradleEnterpriseApiClient.java:121]: ✅ Done

File: components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/api/client/GradleEnterpriseApiClient.java (hunk 2 above)
Comment (jthurne) [reply components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/api/client/GradleEnterpriseApiClient.java]: ✅ Fixed

**Review by etiennestuder — APPROVED** [review]

**Pull request merged**